src/components/queryBuilder/OtelVersionSelect.tsx

```diff
@@ -0,0 +1,65 @@
+import React, { useEffect } from 'react';
+import { SelectableValue } from '@grafana/data';
+import { InlineFormLabel, Select, Switch as GrafanaSwitch, useTheme } from '@grafana/ui';
+import { versions as allVersions } from 'otel';
+import selectors from 'labels';
+
+interface OtelVersionSelectProps {
+  enabled: boolean,
+  onEnabledChange: (enabled: boolean) => void,
+  selectedVersion: string,
+  onVersionChange: (version: string) => void,
+  defaultToLatest?: boolean,
+}
+
+export const OtelVersionSelect = (props: OtelVersionSelectProps) => {
+  const { enabled, onEnabledChange, selectedVersion, onVersionChange, defaultToLatest } = props;
+  const { label, tooltip } = selectors.components.OtelVersionSelect;
+  const options: SelectableValue[] = (allVersions || []).
+    map(v => ({
+      label: `${v.version}${v.name ? (` (${v.name})`) : ''}`,
+      value: v.version
+    }));
+
+    const hasCurrentVersion = allVersions.find(v => v.version === selectedVersion);
```

Comment:
How can it be that an unknown version is selected?

Reply:
I added this since there was similar logic for the column selector, the idea was to show the original/saved value even if it wasn't present. This would happen in the very rare case where someone uses a mismatched version of the plugin, where a newer version dashboard was imported into the older version. Without it, the selection would default to the latest version